### PR TITLE
Add details on adoptedStyleSheets, loading, URL, content-type

### PR DESCRIPTION
**Pull request by rakina**
Adding details on how to use constructed stylesheets to `adoptedStyleSheets`, defining a constructor for StyleSheetList and restricting where it can be used for resolving #23.

Also adding details on what content-type can @imports inside the sheet can handle (only `text/css`) for resolving #14, the URL of the sheet for resolving #10 , and reason of rejecting failed loads.

***
<a href="https://pr-preview.s3.amazonaws.com/WICG/construct-stylesheets/pull/35.html" title="Last updated on Sep 12, 2018, 1:43 AM GMT (0121a7a)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/WICG/construct-stylesheets/35/35e3f88...0121a7a.html" title="Last updated on Sep 12, 2018, 1:43 AM GMT (0121a7a)">Diff</a>

### CODE REVIEW

**Comment by rakina** [comment]
Friendly ping for review @tabatkins

File: index.bs [hunk 1]
```diff
@@ -63,7 +59,8 @@ partial interface DocumentOrShadowRoot {
 			set <var>sheet’s</var> CSS rules to an empty list.
 		6. Let <var>promise</var> be a promise.
 		7. Wait for loading of <a spec=css-cascade-4>@import</a> rules in <var>sheet</var>.
-			* If any of them failed to load, reject <var>promise</var>.
+			* If any of them resulted in a resource with a <a spec=html>Content-Type metadata</a> of anything other than `text/css`, reject <var>promise</var> with a {{TypeError}}.
```
Comment (ewilligers) [inline index.bs]:
```
<code>text/css</code>
```

**Comment by ewilligers** [comment]
To fully resolve https://github.com/WICG/construct-stylesheets/issues/23, should we prevent a constructed style sheet from being added to two documents?

**Comment by rakina** [comment]
> To fully resolve #23, should we prevent a constructed style sheet from being added to two documents?

Under "Using Constructed Stylesheets", there is a line saying a `CSSStyleSheet`can only be applied to the `Document` it is constructed on, or any `ShadowRoot` in the `Document` it is constructed on. Should we make it clearer, or maybe should we throw errors if we add to another `Document`'s `adoptedStyleSheets`?

**Review by ewilligers — APPROVED** [review]
This looks good.

**Comment by ewilligers** [comment]
> Under "Using Constructed Stylesheets", there is a line saying a `CSSStyleSheet`can only be applied to the `Document` it is constructed on, or any `ShadowRoot` in the `Document` it is constructed on. Should we make it clearer, or maybe should we throw errors if we add to another `Document`'s `adoptedStyleSheets`?

Yes, we should make it clear that an exception is thrown if we attempt to assign to `adoptedStyleSheets` any `CSSStyleSheet` not made by `createCSSStyleSheet(text)` or `createEmptyCSSStyleSheet()` calls, or any `CSSStyleSheet` made by calls on a different document.

Currently it is not obvious if an exception occurs or if such style sheets are silently ignored.

**Comment by rakina** [comment]
>Yes, we should make it clear that an exception is thrown if we attempt to assign to adoptedStyleSheets any CSSStyleSheet not made by createCSSStyleSheet(text) or createEmptyCSSStyleSheet() calls, or any CSSStyleSheet made by calls on a different document.
>
> Currently it is not obvious if an exception occurs or if such style sheets are silently ignored.

I see. Added a line detailing `TypeError` is thrown. Thanks!

**Pull request merged**